docs/content/en/latest/references/embed_semantic_search/_index.md

```diff
@@ -0,0 +1,179 @@
+        deepSearch: true,
+        limit: 10,
+        // backend and workspace are optional if provided via context
+    });
+
+    return (
+        <div>
+            <input
+                type="text"
+                value={searchTerm}
+                onChange={(e) => setSearchTerm(e.target.value)}
+                placeholder="Search..."
+            />
+
+            {searchStatus === "loading" && <div>Loading...</div>}
+            {searchStatus === "error" && <div>Error: {searchError}</div>}
+
+            {searchStatus === "success" && (
+                <ul>
+                    {searchResults.map((item) => (
+                        <li key={item.id} onClick={() => handleItemClick(item)}>
+                            {item.title}
+                        </li>
+                    ))}
+                </ul>
+            )}
+        </div>
+    );
+};
+```
+
+### Hook Parameters
+
+| Name        | Type               | Default | Description                                                              |
+| ----------- | ------------------ | ------- | ------------------------------------------------------------------------ |
+| searchTerm  | string             | -       | The search term to use for semantic search                               |
+| objectTypes | GenAIObjectType[]  | []      | A list of object types to search for                                     |
+| deepSearch  | boolean            | false   | Enable deep search to find dashboards by their contents                  |
+| limit       | number             | -       | Maximum number of search results to return                               |
+| backend     | IAnalyticalBackend | -       | Backend instance. Falls back to BackendProvider context if not specified |
+| workspace   | string             | -       | Workspace ID. Falls back to WorkspaceProvider context if not specified   |
+
+### Hook Return Value
+
+The hook returns an object with the following properties:
+
+| Name          | Type                                        | Description                                                                                   |
+| ------------- | ------------------------------------------- | --------------------------------------------------------------------------------------------- |
+| searchStatus  | "idle" \| "loading" \| "error" \| "success" | Current status of the search operation                                                        |
+| searchError   | string                                      | Error message if the search failed                                                            |
+| searchResults | ISemanticSearchResultItem[]                 | Array of search results                                                                       |
+| relationships | ISemanticSearchRelationship[]               | Array of relationships between search results (e.g., dashboard containing a matching insight) |
+
+## Search Result Item Interface
```

Comment:
Since we show this interface here, shouldn't we also show `ISemanticSearchRelationship`?